vulcan/lib/client/jsx/reducers/vulcan_reducer.ts

```diff
@@ -82,7 +82,7 @@ export default function VulcanReducer(state: VulcanState, action: VulcanAction):
       return {
         ...state,
         session: action.session,
-        inputs: {...state.inputs, ...action.session.inputs},
```

Comment:
Are you sure you want to remove the `...state.inputs` bit? I think the original intent was to preserve inputs that you may have entered that have not been saved in the local storage session, yet. I'll play around with it in a bit to observe behavior, but seems like this may change how that works.

Reply:
Can you clarify what you mean by UI confusion issues when you load settings, and the exact behavior you saw? I was able to reproduce something, but putting this back to include `...state.inputs` seemed to fix what I was.

Here are the steps I took that led to UI confusion:

1) Ran the UMAP workflow until the "Color by" option appeared.
2) Loaded up a JSON file with only the default primary inputs, with one value changed for # of UMIs.
3) UI reset back to the three "Record Selection" inputs. Run button was active, and the UI widget showed the three previous values that I had selected.
4) Clicking RUN didn't work. I think because even though the UI widget had values, those values had not been translated into state (overwritten by my JSON upload).
5) Had to reset and then re-select things in the widget to continue on.

Is that what you mean by UI confusion?

However, adding the above back into the reducer allowed me to:


1) Ran the UMAP workflow until the "Color by" option appeared.
2) Loaded up a JSON file with only the default primary inputs, with one value changed for # of UMIs.
3) UI reset back to the three "Record Selection" inputs. Run button was active, and the UI widget showed the three previous values that I had selected.
4) State automatically continued back to the "confirm record selection" step, saw that the same 4 values were present and selected, and continued running the workflow beyond that (to filtering, PCA, etc.), with no additional user interactions.

Not sure if the latter behavior is what you expect, when you upload a new JSON file?